src/components/Items/ItemEditor.vue

```diff
@@ -0,0 +1,173 @@
+      <icon-button
+        class="rounded-tl-xl rounded-bl-xl"
+        :iconConfig="previousItemIconConfig"
+        @click="updateSelectedItemIndex(selectedItemIndex - 1)"
+        :class="{ 'opacity-50': isFirstItem }"
+        :disabled="isFirstItem"
+      ></icon-button>
+
+      <!-- next item button -->
+      <icon-button
+        class="rounded-tr-xl rounded-br-xl"
+        :iconConfig="nextItemIconConfig"
+        @click="updateSelectedItemIndex(selectedItemIndex + 1)"
+        :class="{ 'opacity-50': isLastItem }"
+        :disabled="isLastItem"
+      ></icon-button>
+
+      <!-- add item button -->
+      <icon-button class="rounded-xl" :iconConfig="addItemIconConfig"></icon-button>
+
+      <!-- delete item button -->
+      <icon-button
+        class="rounded-xl bg-delete-button"
+        :iconConfig="deleteItemIconConfig"
+      ></icon-button>
+    </div>
+
+    <!-- item editor -->
+    <div class="h-full border-2 rounded-t-xl mr-2 ml-2 p-2 item-editor-box">
+      <!-- question input box -->
+      <input-text
+        :placeholder="'placeholder'"
+        :title="'Question'"
+        v-model:value="questionText"
+        ref="questionText"
+      ></input-text>
+    </div>
+  </div>
+</template>
+
+<script>
+import IconButton from "../UI/Buttons/IconButton.vue";
+import ItemDropDown from "../UI/DropDownMenu/ItemDropDown.vue";
+import InputText from "../UI/Text/InputText.vue";
+
+export default {
+  name: "ItemEditor",
+
+  data() {
+    return {
+      previousItemIconConfig: {
+        enabled: true,
+        iconName: "chevron-left-solid",
+      },
+      nextItemIconConfig: {
+        enabled: true,
+        iconName: "chevron-right-solid",
+      },
+      addItemIconConfig: {
+        enabled: true,
+        iconName: "plus-solid",
+      },
+      deleteItemIconConfig: {
+        enabled: true,
+        iconName: "delete",
+      },
+      selectedItemIndex: 0,
+    };
+  },
+
+  props: {
+    itemList: {
+      default: () => [],
+      type: Array,
+    },
+  },
+
+  components: {
+    ItemDropDown,
+    IconButton,
+    InputText,
+  },
+
+  created() {},
+
+  methods: {
+    capitalizeFirstLetter(str) {
+      return str.charAt(0).toUpperCase() + str.slice(1);
+    },
+    updateSelectedItemIndex(index) {
+      this.selectedItemIndex = index;
+    },
+  },
+
+  computed: {
+    localItemList: {
+      // a local copy of the item list
+      get() {
+        return this.itemList;
+      },
+      set(localItemList) {
+        this.$emit("update:itemList", localItemList);
+      },
+    },
+    itemOptionsList() {
+      // preparing an options list to pass to the dropdown
+      // TODO: ditch the primevue implementation
+      var optionsList = [];
+      this.localItemList.forEach((item, itemIndex) => {
+        var currentItem = {};
+        currentItem["index"] = itemIndex;
+        var itemType = this.capitalizeFirstLetter(item.type);
+        currentItem["label"] = `${itemType} ${itemIndex + 1}`;
```

Comment:
Do we need to prepare a label here - we can do this in whatever component we want to display in whatever format no? can ditch this process of preparing optionsList separately and directly use it in whatever way needed in the code?

Reply:
Talked on the phone - discarding this comment
The reason: The DropDown component needs a list of labels to show in the drop down. Whatever parent component is calling the dropdown component, it can pass a list of labels (that was prepared in the parent component itself) as a prop. This way, the dropdown component stays independent